### PR TITLE
Friend notes 

### PR DESCRIPTION
**Pull request by Rheon-D**
As per #1877

**ChatInputManager**
Added max characters functionality

**Friend Notes**
Added a new plugin that allows adding/editing notes on friends in the friends list.
- 128 character long notes
- Automagically update on name changes (Providing there is only one change)
- Automagically delete notes for people who are removed

### CODE REVIEW

**Comment by JvckRS** [comment]
Amazing!

**Comment by SoyChai** [comment]
>+ Automagically update on name changes (Providing there is only one change)

I'm so glad you already thought about this, great job!

File: runelite-client/src/main/java/net/runelite/client/plugins/friendnotes/FriendNoteOverlay.java [hunk 1]
```diff
@@ -0,0 +1,71 @@
...
+import net.runelite.client.ui.overlay.tooltip.TooltipManager;
+
+class FriendNoteOverlay extends Overlay
+{
+	private final Client client;
+	private final FriendNotesPlugin plugin;
+	private final TooltipManager tooltipManager;
+
+	@Inject
+	FriendNoteOverlay(Client client, FriendNotesPlugin plugin, TooltipManager tooltipManager)
```
Comment (deathbeam) [inline runelite-client/src/main/java/net/runelite/client/plugins/friendnotes/FriendNoteOverlay.java]: Make this constructor private

File: runelite-client/src/main/java/net/runelite/client/plugins/friendnotes/FriendNotesPlugin.java [hunk 2]
```diff
@@ -0,0 +1,221 @@
...
+			hoveredFriend = null;
+		}
+	}
+
+	@Subscribe
+	public void onMenuOptionClicked(MenuOptionClicked event)
+	{
+		if (WidgetInfo.TO_GROUP(event.getWidgetId()) == WidgetInfo.FRIENDS_LIST.getGroupId())
+		{
+
```
Comment (deathbeam) [inline runelite-client/src/main/java/net/runelite/client/plugins/friendnotes/FriendNotesPlugin.java]: remove newline

File: runelite-client/src/main/java/net/runelite/client/plugins/friendnotes/FriendNotesPlugin.java [hunk 3]
```diff
@@ -0,0 +1,221 @@
...
+					{
+						return;
+					}
+
+					content = Text.removeTags(content).trim();
+					log.debug("Set note for '{}': '{}'", sanitizedTarget, content);
+					setFriendNote(sanitizedTarget, content);
+				});
+			}
+
```
Comment (deathbeam) [inline runelite-client/src/main/java/net/runelite/client/plugins/friendnotes/FriendNotesPlugin.java]: Remove this empty line

File: runelite-client/src/main/java/net/runelite/client/plugins/friendnotes/HoveredFriend.java [hunk 4]
```diff
@@ -0,0 +1,38 @@
...
+ *
+ */
+package net.runelite.client.plugins.friendnotes;
+
+import lombok.AllArgsConstructor;
+import lombok.Getter;
+
+@AllArgsConstructor
+@Getter
+class HoveredFriend
```
Comment (deathbeam) [inline runelite-client/src/main/java/net/runelite/client/plugins/friendnotes/HoveredFriend.java]: Just make this class @Value

**Pull request merged**